Pipelines/templates/dotnet-publish-nix-job.yml

```diff
@@ -0,0 +1,96 @@
+parameters:
+# Job Name
+- name: jobName
+  type: string
+  default: 'publish_nix'
+# Optional Pre-Build Steps
+- name: 'preBuild'
+  type: stepList
+  default: []
+# Version of Dotnet SDK to use
+- name: dotnetVersion
+  type: string
+  default: '3.1.x'
+# Version of NuGet Tool to use
+- name: nugetVersion
+  type: string
+  default: '5.x'
+# Path to .csproj or .sln
+- name: projectPath
+  type: string
+  default: ''
+# Build Configuration
+- name: buildConfiguration
+  type: string
+  default: 'Release'
+# Project Name
+- name: projectName
+  type: string
+  default: ''
+# Release Version
+- name: releaseVersion
+  type: string
+  default: ''
+# Pipeline Artifact Name
+- name: artifactName
+  type: string
+  default: 'Unsigned_Binaries'
+
+jobs:
+- job: ${{ parameters.jobName }}
+  displayName: Dotnet Publish Linux/Mac
+  pool:
+    vmImage: 'ubuntu-latest'
+  steps:
+  - task: UseDotNet@2
+    displayName: Install Dotnet SDK
+    inputs:
+      packageType: 'sdk'
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
+    displayName: Install Nuget Tool
+    inputs:
+      versionSpec: ${{ parameters.nugetVersion }}        
+  - bash: 'echo "##vso[task.setvariable variable=ReleaseVersion;]$RELEASE"'
+    env:
+      RELEASE: ${{ parameters.releaseVersion }}
+    displayName: Set Release Version
+  - ${{ parameters.preBuild }}
```

Comment:
you set the version before you run the prebuild task that sets the version here.  Am I missing something?

Reply:
Doing this to provide multiple ways of setting the release version. Not immediately useful, but may need it in the future.